src/lib/network_pool/transacation_pool_reorg_tests.ml

```diff
@@ -0,0 +1,433 @@
+open Core
+open Async
+open Mina_base
+open Mina_transaction
+open Inline_test_quiet_logs
+open Signature_lib
+open Test_utils
+
+let%test_module "transaction pool reorg tests" =
+( module struct
+  
+
+module Account_spec = struct
+  type t = {
+    key_idx : int
+    ; balance: int
+    ; nonce: int
+  } [@@deriving sexp]
+
+  let rec find_in_test_keys ?(n=0) (x:Keypair.t) =
+    if Keypair.equal test_keys.(n) x then n 
+    else find_in_test_keys x ~n:(n+1)
+
+  let create key_idx balance nonce = 
+    {
+      key_idx
+      ; balance 
+      ; nonce
+    }
+
+  let of_ledger_row ledger_row = 
+    let (kp,balance,nonce,_)  = ledger_row in
+    create 
+       (find_in_test_keys kp) 
+       (Currency.Amount.to_nanomina_int balance)
+       (Unsigned.UInt32.to_int nonce)
+  
+  let apply_payment amount fee t = 
+    create t.key_idx (t.balance - amount - fee) (t.nonce + 1)
+
+  let apply_zkapp fee t = create t.key_idx (t.balance - fee) (t.nonce + 1)
```

Comment:
Are all zkapps in this test created with with zero account updates?

If so, this looks ok, but let's add a comment in the code about this matter.

Reply:
Maybe name is misleading. apply_zkapp function in context of account_spec is only modify spec state which we are using for tracking balance and nonce. So no zkapp command or update is done. Just noting that from this account zkapp was sent and we are deducting fee and increases nonce